appmap/unittest.py

```diff
@@ -12,23 +13,29 @@ def get_test_location(cls, method_name):
     return get_function_location(fn)
 
 
-def run(self, result=None):
-    method_name = self.id().split('.')[-1]
-    # Use the result's location if provided (e.g. by pytest),
-    # otherwise cobble one together ourselves.
-    if hasattr(result, 'location'):
-        location = result.location
-    else:
-        location = get_test_location(self.__class__, method_name)
+# unittest.case._Outcom.testPartExecutor is used by all supported
+# versions of unittest to run test cases. `isTest` will be True when
+# the part is the actual test method, False when it's setUp or
+# teardown.
+@wrapt.patch_function_wrapper('unittest.case', '_Outcome.testPartExecutor')
+@contextmanager
+def testPartExecutor(wrapped, _, args, kwargs):
+    def _args(test_case, *_, isTest=False, **__):
+        return (test_case, isTest)
+    test_case, is_test = _args(*args, **kwargs)
+    if not is_test:
```

Comment:
This gave me pause. Any reason to do it this way instead of using `kwargs.get('isTest')` in the `if` directly, and `test_case = args[0]`? Just wondering, if you find it clearer like this then that's a reason enough :)

Reply:
This way of extracting the arguments is described here: https://wrapt.readthedocs.io/en/latest/decorators.html#processing-function-arguments .

I believe this is necessary to handle the case where the caller of the wrapped function passes the first argument as a kwarg.

Reply:
Ah, ok, that makes sense. Thanks!